index.bs

```diff
@@ -13629,6 +13800,250 @@ internal method as follows.
+        1.  [=list/Append=] [=!=] [$ToString$](|i|) to |keys|.
+        1.  Set |i| to |i| + 1.
+    1.  [=list/Extend=] |keys| with [=!=] |O|.\[[OwnPropertyKeys]]().
+    1.  Return [=!=] [$CreateArrayFromList$](|keys|).
+</div>
+
+<h4 id="es-observable-array-preventExtensions"><code>preventExtensions</code></h4>
+
+<div algorithm="observable array exotic object preventExtensions trap">
+    The steps for the <code>preventExtensions</code> proxy trap for
+    [=observable array exotic objects=] are as follows:
+
+    1.  Return <emu-val>false</emu-val>.
+</div>
+
+<h4 id="es-observable-array-set"><code>set</code></h4>
+
+<div algorithm="observable array exotic object set trap">
+    The steps for the <code>set</code> proxy trap for
+    [=observable array exotic objects=], given |O|, |P|, |V|, and |Receiver|, are as follows:
+
+    1.  Let |handler| be the <emu-val>this</emu-val> value.
+    1.  If |P| is "length", then return the result of
+        [=observable array exotic object/setting the length=] given |handler| and |V|.
+    1.  If |P| [=is an array index=], then return the result of
+        [=observable array exotic object/setting the indexed value=] given |handler|, |P|, and |V|.
+    1.  Return [=?=] |O|.\[[Set]](|P|, |V|, |Receiver|).
+</div>
+
+<h4 id="es-observable-array-abstract-operations">Abstract operations</h4>
+
+<div algorithm>
+    To <dfn for="observable array exotic object" lt="set the length|setting the length">set the length</dfn>
+    of an observable array exotic object given |handler| and |newLen|: 
+
+    1.  Set |newLen| to [=?=] [$ToUint32$](|newLen|).
+    1.  Let |numberLen| be [=?=] [$ToNumber$](|newLen|).
+    1.  If |newLen| ≠ |numberLen|, then throw a {{RangeError}} exception.
+    1.  Let |oldLen| be |handler|.\[[BackingList]]'s [=list/size=].
+    1.  If |newLen| > |oldLen|, then return <emu-val>false</emu-val>.
+    1.  Let |indexToDelete| be |oldLen| &minus; 1.
+    1.  [=While=] |indexToDelete| ≥ |newLen|:
+        1.  Perform the algorithm steps given by |handler|.\[[DeleteAlgorithm]], given
+            |handler|.\[[BackingList]][|indexToDelete|] and |indexToDelete|.
+        1.  [=list/Remove=] the last item from |handler|.\[[BackingList]].
+        1.  Set |indexToDelete| to |indexToDelete| &minus; 1.
+</div>
+
+<div algorithm>
+    To <dfn for="observable array exotic object" lt="set the indexed value|setting the indexed value">set the indexed value</dfn>
+    of an observable array exotic object given |handler|, |P|, and |V|: 
+
+    1.  Let |oldLen| be |handler|.\[[BackingList]]'s [=list/size=].
+    1.  Let |index| be [=!=] [$ToUint32$](|P|).
+    1.  If |index| > |oldLen|, return <emu-val>false</emu-val>.
+    1.  Let |idlValue| be the result of [=converted to an IDL value|converting=]
+        |V| to the type given by |handler|.\[[Type]].
+    1.  If |index| &lt; |oldLen|, then:
+        1.  Perform the algorithm steps given by |handler|.\[[DeleteAlgorithm]], given
```

Comment:
It's worth noting somewhere where the overall concept is defined that a set will do the delete algorithm, then the set algorithm, and only then change the value.  Because a plausible alternative would have been to just do the set algorithm and let the implementation figure out that there's the implicit delete of the old value...  Either choice is OK, I think; we just need to make sure we make it clear to someone who has not read these specific steps which one is happening.